lsp/nls/tests/inputs/completion-fun-parameter-contract.ncl

```diff
@@ -0,0 +1,17 @@
+### /file.ncl
+{
+  Schema = {
+    foo | { bar }
+  },
+  some_func
+    | Schema -> Dyn
+    = fun src =>
+      {
+        # TODO: we shouldn't need to type the first letter to get completion.
+        some_val = src.foo.b
```

Comment:
I realized that for some reason, the completion doesn't fire if we don't start to typing a prefix of an existing field, while this doesn't happen for other kind of completion (for type-based completion for example, if you start to type `std.`, you get a list of all available functions - same for a record literal, if you define `record = {...}` and then later type `record.`, you get a full list - but here we get nothing until we type the first letter of a field). I'll open a separate issue.

Reply:
In fact it's the good old NLS can't handle some incomplete inputs. If I add an identifier at the end of this line, something like `src_val = src.foo.  dummy_value`, which is then parsed correctly (because right now you can have a space between fields and the `.` character, for the best or the worst), then you get completion.